### PR TITLE
[CodeQuality] Add SwitchNegatedTernaryRector

### LINKED ISSUE #4927: [CodeQuality] Add Switch negated ternary Rector
Negated ternary is harder to read than positive one. 

```diff
/**
     * @return Type[]
     */
    private function getTypes(Type $type): array
    {
-        return ! $type instanceof UnionType
+        return $type instanceof UnionType
-            ? [$type]
+            ? $type->getTypes()
-            : $type->getTypes();
+            : [$type];
    }
```

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes #4927

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules/code-quality/src/Rector/Ternary/SwitchNegatedTernaryRector.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+        if ($node->if === null) {
+            return null;
+        }
+
+        $node->cond = $node->cond->expr;
+        $if = $node->if;
+        $node->if = $node->else;
+
+        $node->else = $if;
```
Comment (TomasVotruba) [inline rules/code-quality/src/Rector/Ternary/SwitchNegatedTernaryRector.php:78]: Here this trick can be used:

```suggestion
        [$node->if, $node->else] = [$node->else, $node->if];
```

Comment (samsonasik) [reply rules/code-quality/src/Rector/Ternary/SwitchNegatedTernaryRector.php]: updated.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by samsonasik** [comment]
rebased.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Comment by TomasVotruba** [comment]
Could you add it to `code-quality.php` set?

**Comment by samsonasik** [comment]
I will try.

**Pull request merged**